### PR TITLE
It supports have_enqueud_mail matcher sub class of DeliveryJob.

### PR DESCRIPTION
**Pull request by yalab**
Hi.

I Fixed a bug.

Thanks.

### CODE REVIEW

File: spec/rspec/rails/matchers/have_enqueued_mail_spec.rb [hunk 1]
```diff
@@ -21,6 +21,15 @@ class UnifiedMailer < ActionMailer::Base
       def test_email; end
       def email_with_args(arg1, arg2); end
     end
+
+    class SubClassDeliveryJob < ActionMailer::DeliveryJob
```
Comment (JonRowe) [inline spec/rspec/rails/matchers/have_enqueued_mail_spec.rb]:
```suggestion
    class DeliveryJobSubClass < ActionMailer::DeliveryJob
```

File: spec/rspec/rails/matchers/have_enqueued_mail_spec.rb [hunk 2]
```diff
@@ -21,6 +21,15 @@ class UnifiedMailer < ActionMailer::Base
...
+    end
+
+    class SubClassUnifiedMailer < ActionMailer::Base
+      self.delivery_job = SubClassDeliveryJob
```
Comment (JonRowe) [inline spec/rspec/rails/matchers/have_enqueued_mail_spec.rb]:
```suggestion
      self.delivery_job = DeliveryJobSubClass
```

File: spec/rspec/rails/matchers/have_enqueued_mail_spec.rb [hunk 3]
```diff
@@ -21,6 +21,15 @@ class UnifiedMailer < ActionMailer::Base
       def test_email; end
       def email_with_args(arg1, arg2); end
     end
+
+    class SubClassDeliveryJob < ActionMailer::DeliveryJob
+    end
+
+    class SubClassUnifiedMailer < ActionMailer::Base
```
Comment (JonRowe) [inline spec/rspec/rails/matchers/have_enqueued_mail_spec.rb]:
```suggestion
    class UnifiedMailerWithDeliveryJobSubClass < ActionMailer::Base
```

**Comment by yalab** [comment]
> an interesting use of the <= operator, is that documented anywhere?

See `ri 'Class#<=' on your terminal.

```
= Class#<=

(from ruby core)
=== Implementation from Module
------------------------------------------------------------------------
  mod <= other   ->  true, false, or nil

------------------------------------------------------------------------

Returns true if mod is a subclass of other or is the
same as other. Returns nil if there's no relationship between
the two. (Think of the relationship in terms of the class definition:
"class A < B" implies "A < B".)
```

**Pull request merged**